### ISSUE TITLE
"Protocol version not supported" when Client connects to Server when using banner property on Server

### ISSUE THREAD

**Issue by sivo** [mscdex/ssh2#362]
When using the first server example [here](https://github.com/mscdex/ssh2#server-examples), and the first client example [here](https://github.com/mscdex/ssh2#client-examples), an error occurs when the `banner` property is set to a string in the server: "Protocol version not supported"

When using `debug: console.log` on the client:
```
DEBUG: Local ident: 'SSH-2.0-ssh2js0.0.19'
DEBUG: Client: Trying 127.0.0.1 on port 2222 ...
DEBUG: Client: Connected
DEBUG: Parser: IN_INIT
DEBUG: Parser: IN_GREETING
DEBUG: Parser: IN_HEADER
DEBUG: Parser: protocol version not supported: SSH
error  Protocol version not supported
```

And on the server:
```
[6234.15905622] DEBUG: Local ident: 'SSH-2.0-ssh2js0.0.19srv'
[6234.15905622] DEBUG: Parser: IN_INIT (waiting for identification begin)
[6234.15905622] DEBUG: Parser: IN_INIT (waiting for rest of identification)
[6234.15905622] DEBUG: Parser: IN_GREETING
[6234.15905622] DEBUG: Parser: IN_HEADER
[6234.15905622] DEBUG: Remote ident: 'SSH-2.0-ssh2js0.0.19'
[6234.15905622] DEBUG: Outgoing: Writing KEXINIT
Client connected!
[6234.15905622] DEBUG: Parser: IN_PACKETBEFORE (expecting 8)
Client disconnected
```

Using version 0.4.12.

Note: I did change to using password authentication on client and server, but I don't think that would affect the problem

**Comment by codebudo** [comment]
I can confirm this behavior. However, I was able to connect using the sftp command line client `sftp -oPort=2223 foo@localhost:/tmp/foo.txt`. There are other problems at this point, but the client does pass the 'protocol' phase and properly authenticate.

**Comment by mscdex** [comment]
This should be fixed with mscdex/ssh2-streams@f1b409ec6. Reinstall `ssh2` and the new `ssh2-streams` with this change should be picked up.

**Comment by codebudo** [comment]
Also confirmed. Thanks @mscdex !
